controllers/module_reconciler.go

```diff
@@ -82,13 +85,13 @@ func (r *ModuleReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 
 	logger := log.FromContext(ctx)
 
-	mod := ootov1alpha1.Module{}
-
-	if err := r.Client.Get(ctx, req.NamespacedName, &mod); err != nil {
-		logger.Error(err, "Could not get module")
-		return res, err
+	mod, err := r.getRequestedModule(ctx, req.NamespacedName)
```

Comment:
better to have it in a separate function: makes code easier to read, and makes changing the implementation easier, in case we want to add some functionality (check) to the function